src/test/java/htsjdk/samtools/SamFilesTest.java

```diff
@@ -58,6 +60,15 @@ public void testFindIndexForSuffixes(final String dataFileSuffix, final String i
 
         Assert.assertNotNull(foundIndexFile);
         Assert.assertTrue(foundIndexFile.getName().endsWith(expectIndexSuffix));
+
+        final Path foundIndexPath = SamFiles.findIndex(dataFile.toPath());
+        if (expectIndexSuffix == null) {
+            Assert.assertNull(foundIndexPath);
+            return;
```

Comment:
Refactor out these `return` statements from this test case, replacing with `else` blocks -- otherwise when `expectIndexSuffix == null` you'll only test the `File` case and not the `Path` case